plugins/tplink/tl_sg108e

```diff
@@ -0,0 +1,210 @@
+
+  [tl_sg108e]
+  env.switch 192.168.1.12
+  env.username admin
+  env.password mySecretPassword
+  env.p1 'Link to PC1'
+  env.p2 'Link to server1'
+  env.p3 'Not used'
+  env.p4 'Link to AP'
+  env.p5 'Link to PC2'
+  env.p6 'Link to PC3'
+  env.p7 'Not used'
+  env.p8 'Uplink'
+
+Requires WWW:Mechanize module:
+  sudo apt-get install libwww-mechanize-perl
+
+=head1 BUGS/GOTCHAS
+
+The link speed is represented as a number:
+  0 - down
+  5 - 100Mbps full
+  6 - 1Gbps
+
+=head1 AUTHOR
+
+Adrian Popa (https://github.com/mad-ady)
+
+=head1 COPYRIGHT
+
+Copyright (c) 2018, Adrian Popa
+
+All rights reserved. This program is free software; you can
+redistribute it and/or modify it under the terms of the GNU General
+Public License as published by the Free Software Foundation; version 2
+dated June, 1991.
+
+=head1 VERSION
+
+  $Id$
+
+=cut
+
+# read parameters from munin
+my $switch   = ( $ENV{switch}   || '192.168.1.1' );
+my $username = ( $ENV{username} || 'admin' );
+my $password = ( $ENV{password} || 'admin' );
+my $numberOfPorts = 8;
+
+#populate the ports and descriptions based on ENV
+my %ports = ();
+for ( 1 .. $numberOfPorts ) {
+    my $i = $_;
+    if ( defined $ENV{"p$i"} ) {
+        $ports{$i} = $ENV{"p$i"};
+    }
+    else {
+        #no description
+        $ports{$i} = "";
+    }
+}
+
+if ( $ARGV[0] and $ARGV[0] eq 'autoconf' ) {
+    print "no (manual configuration needed)\n";
+    exit 0;
+}
+
+if ( $ARGV[0] and $ARGV[0] eq "config" ) {
+    foreach my $graphType (qw/Packets Speed/) {
+        foreach my $port ( sort keys %ports ) {
+            print "multigraph ${graphType}_if_$port\n",
+              "graph_title $graphType for $switch port $port $ports{$port}\n",
+              "graph_info $graphType graph for port $port $ports{$port}\n",
+              "graph_args --base 1000 -l 0\n",
+              "graph_scale yes\n",
+              "graph_category switch\n";
+            if ( $graphType eq 'Speed' ) {
+                print "graph_vlabel speed\n";
+                print "p${port}.label Port $port Link speed\n";
+                print "p${port}.type GAUGE\n";
+            }
+            else {
+                print "graph_vlabel packets\n";
+                foreach my $gb (qw/good bad/) {
+                    foreach my $direction (qw/tx rx/) {
+                        print "p${port}${direction}${gb}.label Port $port $direction ($gb)\n";
+                        print "p${port}${direction}${gb}.type COUNTER\n";
+                    }
+                }
+            }
+        }
+    }
+    exit 0;
+}
+
+# extract data from the switch - uses web scraping (tested with 1.0.2 Build 20160526 Rel.34615)
+
+# print STDERR "Connecting to $switch with user $username";
+my $mech = WWW::Mechanize->new(
+    autocheck             => 0,
+    requests_redirectable => [ 'GET', 'HEAD', 'POST' ],
+    user_agent => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0'
```

Comment:
Is it really necessary, to lie about our user agent?
If yes: please add a comment.

Reply:
I will check, otherwise I'll remove it.